src/request.rs

```diff
@@ -87,19 +90,28 @@ impl<Endpoint> CoapRequest<Endpoint> {
 
     /// Returns the flag in the Observe option.
     pub fn get_observe_flag(&self) -> Option<ObserveOption> {
```

Comment:
Since getting that flag is inherently fallible and you added the fallible `try_get_observe_flag` method, do you think it would make sense to remove this one that just swallows errors?

Reply:
Yes that was my preference but didn't want to break the API.  Per your comments though let's go ahead and do it.